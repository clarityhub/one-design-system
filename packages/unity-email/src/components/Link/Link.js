import React from 'react';
import { node, string } from 'prop-types';

const Link = ({ children, href }) => (
	<a href={href}>{children}</a>
);

Link.propTypes = {
	children: node,
	href: string.isRequired,
};

export default Link;
