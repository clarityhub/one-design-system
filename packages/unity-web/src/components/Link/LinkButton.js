import React from 'react';
import { node, shape, func, string } from 'prop-types';

import Button from '../Buttons/Button';

const LinkButton = ({ history, to, onClick, ...rest }) => (
	<Button
		{...rest}
		onClick={(event) => {
			onClick && onClick(event);
			history.push(to);
		}}
	/>
);

LinkButton.propTypes = {
	children: node.isRequired,
	history: shape({
		push: func.isRequired,
	}).isRequired,
	to: string.isRequired,
};

export default LinkButton;