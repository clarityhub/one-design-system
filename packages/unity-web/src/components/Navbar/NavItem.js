import isPropValid from '@emotion/is-prop-valid';
import styled from '@emotion/styled';
import { css } from '@emotion/core';
import colors from '@clarityhub/unity-core/lib/colors';

const NavItem = styled('div', {
	shouldForwardProp: prop => isPropValid(prop) && !['selected'].includes(prop),
})`
    a {
        color: ${colors.white.default};

        ${({ selected }) => selected && css`
            font-weight: bold;
        `}
    }

    &:not(:first-of-type) {
        margin-left: 3rem;
    }

    /* Used for avatars in the navbar */
    img {
        margin-top: 4px;
    }
`;

export default NavItem;
