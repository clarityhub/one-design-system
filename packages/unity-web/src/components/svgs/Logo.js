import React from 'react';

const Logo = () => (
	<svg width="392px" height="313px" viewBox="0 0 392 313" version="1.1" xmlns="http://www.w3.org/2000/svg">
		<title>Clarity Hub</title>
		<defs></defs>
		<g id="Logo" stroke="none" strokeWidth="1" fill="none" fillRule="evenodd">
			<g id="Simple-Reversed-Filled" transform="translate(-14.000000, -54.000000)" fill="#FFFFFF">
				<g id="Simple" transform="translate(14.000000, 54.000000)">
					<g id="Body">
						<path d="M19,0 L373,0 C383.49341,-1.92760819e-15 392,8.50658975 392,19 L392,294 C392,304.49341 383.49341,313 373,313 L19,313 C8.50658975,313 1.28507213e-15,304.49341 0,294 L0,19 C-1.28507213e-15,8.50658975 8.50658975,1.92760819e-15 19,0 Z M94.3110522,219.890145 C131.145116,249.95077 164.584555,268.451335 195.052311,275.39184 C225.097126,268.451335 258.536566,249.95077 295.370629,219.890145 C348.341694,176.659962 363.445819,112.032788 345.398601,79.8638289 C326.474838,46.1324376 286.428443,38.5339668 243.286713,55.1533026 C219.720716,64.2315483 203.572092,81.1892036 194.840841,106.026269 C186.10959,81.1892036 169.960966,64.2315483 146.394968,55.1533026 C103.253238,38.5339668 63.2068437,46.1324376 44.2830802,79.8638289 C26.2358624,112.032788 41.3399879,176.659962 94.3110522,219.890145 Z"
							id="Combined-Shape"></path>
					</g>
					<g id="Eyes" transform="translate(81.000000, 90.000000)">
						<ellipse id="Oval" cx="34" cy="34.1555017" rx="34" ry="34.1555017"></ellipse>
						<ellipse id="Oval" cx="193" cy="34.1555017" rx="34" ry="34.1555017"></ellipse>
					</g>
					<g id="Nose" transform="translate(176.208469, 155.861224)">
						<path d="M20.4739975,0 C6.82466584,18.3562632 0,32.7064293 0,43.0504983 C0,53.3945674 6.82466584,58.6335169 20.4739975,58.7673469 C32.758396,58.7673469 38.9005953,53.5283974 38.9005953,43.0504983 C38.9005953,32.5725993 32.758396,18.2224332 20.4739975,0 Z"
							id="Line-2"></path>
					</g>
				</g>
			</g>
		</g>
	</svg>
);

export default Logo;
