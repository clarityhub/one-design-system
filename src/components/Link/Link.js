import React from 'react';
import { func, oneOfType, node, string } from 'prop-types';

import Typography from '../Typography';

const Link = ({ component = 'a', children, variant = 'text', color = 'primary', type, ...props }) => (
    <Typography
        type={type ? type : 'link'}
        variant={variant}
        color={color}
        component={component}
        {...props}
    >
        {children}
    </Typography>
);

/**
 * Hack for docz and react-docgen
 */
const LinkPropTypes = () => <div />;
LinkPropTypes.propTypes = {
    component: oneOfType([string, func]),
    children: node,
    variant: string,
    color: string,
    type: string,
};
Link.propTypes = LinkPropTypes.propTypes;
export { LinkPropTypes };

export default Link;
