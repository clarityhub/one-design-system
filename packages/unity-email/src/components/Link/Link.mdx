---
menu: Components
name: Link
---

import { Playground, PropsTable } from 'docz';
import Link from './Link';

# Email Link

```js
import Link from '@clarityhub/unity-email/lib/components/Link';
```

## Prop Types

<PropsTable of={Link} />
