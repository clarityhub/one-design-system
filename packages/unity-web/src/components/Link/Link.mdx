---
menu: Components
name: Link
---

import { Playground, PropsTable } from 'docz';
import Link, { LinkPropTypes } from './Link';

# Link

## \<Link /\> component

```js
import Link from '@clarityhub/unity-web/lib/components/Link';
```

## Prop Types

<PropsTable of={LinkPropTypes} />

## Examples

<Playground>
    <Link href="#">Link</Link>
    <Link type="blockLink" href="#">Block Link</Link>
    <Link href="#" component='div'>Custom Component</Link>
</Playground>