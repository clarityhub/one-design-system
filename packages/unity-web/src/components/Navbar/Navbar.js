import React, { Fragment } from 'react';
import styled from '@emotion/styled';
import css from '@emotion/css';
import { bool, func, node, number, string, oneOfType, element } from 'prop-types';
import colors from '@clarityhub/unity-core/lib/colors';

import Link from '../Link';
import Typography from '../Typography';
import { NAVBAR_HEIGHT } from './config';

const NavbarContainer = styled.nav`
    background: ${colors.notification.default};
    color: ${colors.white.default};
    height: ${NAVBAR_HEIGHT}rem;
    padding-top: 0;
`;

const NavbarContent = styled.div`
    height: ${NAVBAR_HEIGHT}rem;
    display: flex;
    line-height: 1;
    margin: 0 auto;
    padding: 0 3rem;
    justify-items: center;
    align-items: center;

    ${({ maxWidth }) => maxWidth && css`
        max-width: ${maxWidth}px;

        @media (max-width: ${maxWidth + 40}px) {
            width: auto;
        }
    `}
`;

const NavbarBrandWrapper = styled.section`
    a {
        align-items: center;
        display: flex;
        justify-content: center;
        margin-bottom: 0;
        margin-right: 40px;
        text-align: center;
        width: auto;
        
        border-bottom-color: transparent;
        text-decoration: none;

        &:visited {
            border-bottom-color: transparent;
        }

        &:hover {
            border-bottom-color: transparent;
        }
    }

    span {
      font-size: 1rem;
      line-height: 2rem;
      margin-left: 15px;

       @media (max-width: 480px) {
           display: none;
       }
    }

    img,
    svg  {
      height: 40px;
      width: auto;
    }
`;

const Gutter = styled.div`
    width: 2rem;
`;

const NavbarItems = styled.div`
    display: flex;
    flex: 1;
    flex-direction: row;
    justify-items: center;
    align-items: center;
    justify-content: flex-end;
`;

const defaultNavBrandRenderer = ({ path, children }) => (
	<Link href={path}>{children}</Link>
);

const NavbarBrand = ({
	navBrandPath,
	navBrandRenderer,
	navBrandText,
	navBrandLogo: Logo,
}) => (
	<NavbarBrandWrapper>
		{navBrandRenderer({
			path: navBrandPath,
			children: (
				<Fragment>
					<Logo />
					<Typography variant="heading" color="white" component="span">
						{navBrandText}
					</Typography>
				</Fragment>
			),
		})}
	</NavbarBrandWrapper>
);

const Navbar = ({
	children,
	maxWidth = false,
	navBrandText = 'Clarity Hub',
	navBrandLogo,
	navBrandPath = '/',
	navBrandRenderer = defaultNavBrandRenderer,
}) => (
	<NavbarContainer>
		<NavbarContent maxWidth={maxWidth}>
			<NavbarBrand
				navBrandLogo={navBrandLogo}
				navBrandPath={navBrandPath}
				navBrandRenderer={navBrandRenderer}
				navBrandText={navBrandText}
			/>
			<Gutter />

			<NavbarItems>
				{children}
			</NavbarItems>
		</NavbarContent>
	</NavbarContainer>
);

Navbar.propTypes = {
	children: node,
	maxWidth: oneOfType([number, bool]),
	navBrandLogo: oneOfType([node, func, element]).isRequired,
	navBrandPath: string,
	navBrandRenderer: func,
	navBrandText: string,
};

export default Navbar;
