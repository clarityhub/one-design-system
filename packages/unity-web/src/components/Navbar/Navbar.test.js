import React from 'react';
import renderer from 'react-test-renderer';
import Logo from '../svgs/Logo';
import Link from '../Link';
import Navbar from './Navbar';
import NavItem from './NavItem';

describe('<Navbar />', () => {
	it('renders nav items', () => {

		const node = (
			<Navbar
				navBrandText="Dashboard"
				navBrandLogo={Logo}
				maxWidth={500}
			>
				<NavItem selected>
					<Link type="blockLink" href="#">Dashboard</Link>
				</NavItem>
				<NavItem>
					<Link type="blockLink" href="#">Integrations</Link>
				</NavItem>
			</Navbar>
		);
		const tree = renderer.create(node).toJSON();

		expect(tree).toMatchSnapshot();
	});
});