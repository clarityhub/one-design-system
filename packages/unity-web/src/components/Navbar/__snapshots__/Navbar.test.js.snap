// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> renders nav items 1`] = `
.emotion-16 {
  background: #FF6A5C;
  color: #FFF;
  height: 5.3rem;
  padding-top: 0;
}

.emotion-14 {
  height: 5.3rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  line-height: 1;
  margin: 0 auto;
  padding: 0 3rem;
  justify-items: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  max-width: 500px;
}

@media (max-width:540px) {
  .emotion-14 {
    width: auto;
  }
}

.emotion-2 a {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-bottom: 0;
  margin-right: 40px;
  text-align: center;
  width: auto;
  border-bottom-color: transparent;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-2 a:visited {
  border-bottom-color: transparent;
}

.emotion-2 a:hover {
  border-bottom-color: transparent;
}

.emotion-2 span {
  font-size: 1rem;
  line-height: 2rem;
  margin-left: 15px;
}

@media (max-width:480px) {
  .emotion-2 span {
    display: none;
  }
}

.emotion-2 img,
.emotion-2 svg {
  height: 40px;
  width: auto;
}

.emotion-1 {
  border-bottom: 1px solid #566ad2;
  color: #566ad2;
  display: inline-block;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: 'Open Sans',Arial,sans-serif;
  font-weight: 400;
  string: font-family:'Open Sans',Arial,sans-serif;
  font-weight: 400;
  color: #566ad2;
}

.emotion-1:hover,
.emotion-1:active {
  border-bottom: 2px solid #566ad2;
  border-bottom-width: 2px;
  color: #566ad2;
  margin-bottom: -1px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-1:visited {
  border-bottom: 1px solid rgb(50,72,187);
  color: rgb(50,72,187);
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-1:visited:hover,
.emotion-1:visited:active {
  margin-bottom: 0;
}

.emotion-0 {
  font-family: 'Open Sans',Arial,sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  font-family: Futura,'Trebuchet MS',Arial,sans-serif;
  font-weight: 400;
  string: font-family:Futura,'Trebuchet MS',Arial,sans-serif;
  font-weight: 400;
  color: #FFF;
}

.emotion-0 + .emotion-0 {
  margin-top: 1.6rem;
}

.emotion-0 strong,
.emotion-0 b {
  font-weight: 800;
}

.emotion-0 i,
.emotion-0 em {
  font-style: italic;
}

.emotion-4 {
  width: 2rem;
}

.emotion-12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  justify-items: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.emotion-7 a {
  color: #FFF;
  font-weight: bold;
}

.emotion-7:not(:first-of-type) {
  margin-left: 3rem;
}

.emotion-7 img {
  margin-top: 4px;
}

.emotion-6 {
  -khtml-user-select: none;
  -webkit-tap-highlight-color: rgba(0,0,0,0);
  -webkit-touch-callout: none;
  border: 0 !important;
  color: #FDFDFD;
  cursor: pointer;
  display: block;
  font-size: 0.875rem;
  line-height: 1.625rem;
  position: relative;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: color 0.3s;
  transition: color 0.3s;
  -webkit-transition-property: color;
  transition-property: color;
  -webkit-letter-spacing: 0.8px !important;
  -moz-letter-spacing: 0.8px !important;
  -ms-letter-spacing: 0.8px !important;
  letter-spacing: 0.8px !important;
  font-family: 'Open Sans',Arial,sans-serif;
  font-weight: 400;
  string: font-family:'Open Sans',Arial,sans-serif;
  font-weight: 400;
  color: #566ad2;
}

.emotion-6:visited {
  color: #FDFDFD !important;
}

.emotion-6::before {
  background: #FDFDFD;
  bottom: -6px;
  content: '';
  height: 2px;
  left: 0;
  position: absolute;
  -webkit-transform: scale3d(0,0.4,1);
  -ms-transform: scale3d(0,0.4,1);
  transform: scale3d(0,0.4,1);
  -webkit-transition: -webkit-transform 0.1s;
  -webkit-transition: transform 0.1s;
  transition: transform 0.1s;
  width: 100%;
}

.emotion-6.active::before,
.emotion-6:hover::before,
.emotion-6:focus::before {
  -webkit-transform: scale3d(0.8,0.4,1);
  -ms-transform: scale3d(0.8,0.4,1);
  transform: scale3d(0.8,0.4,1);
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
  -webkit-transition-timing-function: cubic-bezier(0.4,0,0.2,1);
  transition-timing-function: cubic-bezier(0.4,0,0.2,1);
}

.emotion-10 a {
  color: #FFF;
}

.emotion-10:not(:first-of-type) {
  margin-left: 3rem;
}

.emotion-10 img {
  margin-top: 4px;
}

<nav
  className="emotion-16 emotion-17"
>
  <div
    className="emotion-14 emotion-15"
  >
    <section
      className="emotion-2 emotion-3"
    >
      <a
        className="emotion-1"
        href="/"
      >
        <svg
          height="313px"
          version="1.1"
          viewBox="0 0 392 313"
          width="392px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title>
            Clarity Hub
          </title>
          <defs />
          <g
            fill="none"
            fillRule="evenodd"
            id="Logo"
            stroke="none"
            strokeWidth="1"
          >
            <g
              fill="#FFFFFF"
              id="Simple-Reversed-Filled"
              transform="translate(-14.000000, -54.000000)"
            >
              <g
                id="Simple"
                transform="translate(14.000000, 54.000000)"
              >
                <g
                  id="Body"
                >
                  <path
                    d="M19,0 L373,0 C383.49341,-1.92760819e-15 392,8.50658975 392,19 L392,294 C392,304.49341 383.49341,313 373,313 L19,313 C8.50658975,313 1.28507213e-15,304.49341 0,294 L0,19 C-1.28507213e-15,8.50658975 8.50658975,1.92760819e-15 19,0 Z M94.3110522,219.890145 C131.145116,249.95077 164.584555,268.451335 195.052311,275.39184 C225.097126,268.451335 258.536566,249.95077 295.370629,219.890145 C348.341694,176.659962 363.445819,112.032788 345.398601,79.8638289 C326.474838,46.1324376 286.428443,38.5339668 243.286713,55.1533026 C219.720716,64.2315483 203.572092,81.1892036 194.840841,106.026269 C186.10959,81.1892036 169.960966,64.2315483 146.394968,55.1533026 C103.253238,38.5339668 63.2068437,46.1324376 44.2830802,79.8638289 C26.2358624,112.032788 41.3399879,176.659962 94.3110522,219.890145 Z"
                    id="Combined-Shape"
                  />
                </g>
                <g
                  id="Eyes"
                  transform="translate(81.000000, 90.000000)"
                >
                  <ellipse
                    cx="34"
                    cy="34.1555017"
                    id="Oval"
                    rx="34"
                    ry="34.1555017"
                  />
                  <ellipse
                    cx="193"
                    cy="34.1555017"
                    id="Oval"
                    rx="34"
                    ry="34.1555017"
                  />
                </g>
                <g
                  id="Nose"
                  transform="translate(176.208469, 155.861224)"
                >
                  <path
                    d="M20.4739975,0 C6.82466584,18.3562632 0,32.7064293 0,43.0504983 C0,53.3945674 6.82466584,58.6335169 20.4739975,58.7673469 C32.758396,58.7673469 38.9005953,53.5283974 38.9005953,43.0504983 C38.9005953,32.5725993 32.758396,18.2224332 20.4739975,0 Z"
                    id="Line-2"
                  />
                </g>
              </g>
            </g>
          </g>
        </svg>
        <span
          className="emotion-0"
        >
          Dashboard
        </span>
      </a>
    </section>
    <div
      className="emotion-4 emotion-5"
    />
    <div
      className="emotion-12 emotion-13"
    >
      <div
        className="emotion-7 emotion-8"
      >
        <a
          className="emotion-6"
          href="#"
        >
          Dashboard
        </a>
      </div>
      <div
        className="emotion-10 emotion-8"
      >
        <a
          className="emotion-6"
          href="#"
        >
          Integrations
        </a>
      </div>
    </div>
  </div>
</nav>
`;
